src/lib.rs

```diff
@@ -87,8 +86,19 @@ pub fn init() -> ExecutorBuilder<NeedsStrategy> {
     }
 }
 
+/// A context object to be passed into [`execute()`] containing
+/// metadata that allows the caller to fine tune strategies
+#[derive(Debug, Clone, Copy)]
+pub struct ExecuteContext {
+    /// The chance of blocking a future for a significant amount of time with this work
+    pub chance_of_blocking: ChanceOfBlocking,
+    /// A namespace for this operation
+    pub namespace: &'static str,
+}
+
```

Comment:
this needs to be nonexhaustive or we won't be able to add fields in the future

Reply:
you can avoid making a builder by having a `new` method, a `new_with_namespace` method, etc. if you want